### PR TITLE
ItPodTemplates test conversion to Junit5

### PR DESCRIPTION
**Pull request by marinakog**
Latest Result
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1142/

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+      String managedServerPodName = domain2Uid + "-" + MANAGED_SERVER_NAME_BASE + "1";
+      V1Pod managedServerPod = Kubernetes.getPod(domain2Namespace, null, managedServerPodName);
+      assertNotNull(managedServerPod,"The admin pod does not exist in namespace " + domain2Namespace);
+      String serverName = managedServerPod
+          .getMetadata().getLabels()
+          .get("servername");
+      assertNotNull(serverName, "Can't find label servername");
+      assertTrue(serverName.equalsIgnoreCase("managed-server1"),
+          "Can't find or match label servername");
+      String domainName = managedServerPod
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java:158]: throughout the test add some logs and comments as to what you are verifying.

File: new-integration-tests/src/test/resources/wdt-models/wdt-model.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          ListenAddress: None
+          ListenPort: '@@PROP:T3_CHANNEL_PORT@@'
+          PublicAddress: '@@PROP:T3_PUBLIC_ADDRESS@@'
+          PublicPort: '@@PROP:T3_CHANNEL_PORT@@'
+  ServerTemplate:
+    template1:
+      Cluster: '@@PROP:CLUSTER_NAME@@'
+      ListenPort: '@@PROP:MANAGED_SERVER_PORT@@'
+resources:
+  JDBCSystemResource:
```
[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/resources/wdt-models/wdt-model.yaml]: removed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+
+
+    // docker login and push image to docker registry if necessary
+    dockerLoginAndPushImageToRegistry(wdtImage);
+
+    return wdtImage;
+  }
+
+  //create domain from provided image and verify it's start
+  private static void createAndVerifyDomain(String miiImage,
```
[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java]: corrected var name

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+
+    // create encryption secret
+    logger.info("Create encryption secret");
+    String encryptionSecretName = "encryptionsecret";
+    assertDoesNotThrow(() -> createSecretWithUsernamePassword(encryptionSecretName, namespace,
+        "weblogicenc", "weblogicenc"),
+        String.format("create encryption secret failed for %s", encryptionSecretName));
+
+    // create domain and verify
+    logger.info("Create model in image domain {0} in namespace {1} using docker image {2}",
```
[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java]: fixed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+   *                   variables
+   */
+  @Test
+  @DisplayName("Test pod templates using all the variables for domain in image.")
+  public void testPodTemplateUsingVariablesDomainInImage() throws Exception {
+    wdtImage = createAndVerifyDomainInImage();
+    try {
+      logger.info("Create wdt domain and verify that it's running");
+      createAndVerifyDomain(wdtImage, domain2Uid, domain2Namespace, "Image", replicaCount);
+      String managedServerPodName = domain2Uid + "-" + MANAGED_SERVER_NAME_BASE + "1";
```
[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java [hunk 6]
```diff
@@ -0,0 +1,495 @@
...
+          .get("domainuid");
+      assertNotNull(domainuid, "Can't find label domainuid");
+      assertTrue(domainuid.equalsIgnoreCase(domain1Uid),
+          "Can't find expected value for label domainuid");
+
+      String loghome = managedServerPod
+          .getMetadata().getAnnotations()
+          .get("loghome");
+      assertNotNull(loghome, "Can't find label loghome");
+      //value is not initialized since logHomeEnable = false
```
[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java]: by default crd has logHomeEnable = false, so even pod template has LOG_HOME used for annotation it does not get initialized. I left like that to check that it does not prevent pod from running.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java [hunk 7]
```diff
@@ -0,0 +1,495 @@
...
+  }
+
+  @AfterAll
+  public void tearDownAll() {
+
+    // shutdown domain1
+    logger.info("Shutting down domain1");
+    assertTrue(shutdownDomain(domain1Uid, domain1Namespace),
+        String.format("shutdown domain %s in namespace %s failed", domain1Uid, domain1Namespace));
+    if (wdtImage != null) {
```
[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java]: removed image delete (this is custom image created I thought we must delete them in the test)

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java [hunk 8]
```diff
@@ -0,0 +1,495 @@
...
+    // create image with model files
+    logger.info("Create image with model file and verify");
+    String appPath = String.format("%s/../src/integration-tests/apps/testwebapp.war", ITTESTS_DIR);
+
+    List<String> appList = new ArrayList();
+    appList.add(appPath);
+
+    int t3ChannelPort = getNextFreePort(31600, 32767);  // the port range has to be between 31,000 to 32,767
+
+    Properties p = new Properties();
```
[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java:257]: it is used for passing the property to model file, I removed it, we don't need t3 access in this test

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java [hunk 9]
```diff
@@ -0,0 +1,495 @@
...
+   * $(DOMAIN_HOME), $(LOG_HOME) and $(CLUSTER_NAME)
+   * in serverPod for Model In Image domain. Make sure the domain comes up
+   * successfully.
+   *
+   * @throws Exception when the domain crd creation fails or when updating the serverPod with
+   *                   variables
+   */
+  @Test
+  @DisplayName("Test pod templates using all the variables for model in image domain.")
+  public void testPodTemplateUsingVariablesModelInImage() throws Exception {
```
[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java:227]: should be same, sent this question to Ryan, if not needed will remove one of them

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java (hunk 1 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java]: added

**Comment by marinakog** [comment]
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1142 latest results

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java]: also switched to generic wit image

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java:142]: I don't think its needed as there is no difference in the way the template variables are used. Can you remove this test.

**Comment by marinakog** [comment]
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1171/ latest run

**Pull request merged**